src/rospkg/environment.py

```diff
@@ -109,7 +109,7 @@ def get_ros_package_path(env=None):
     """
     if env is None:
         env = os.environ
-    return env.get(ROS_PACKAGE_PATH, None)
+    return env.get(ROS_PACKAGE_PATH, '') + ':/usr/share'
```

Comment:
> Since there behavior must be the same

Must it? In the proposal, `rospkg` will act like the debian version of `rospack` when the env variable is not set and like the packages.ros.org version of `rospack` when it is. I see no reason to have `rospack` look for the env variable and change behavior like `rospkg`.

> While that package can depend on a new version number of rospkg there is no way to prevent users to update only rospkg but not that ros package. The result will be that on Ubuntu the user now has /usr/share in his RPP which will significantly increase the crawling time. This will also be the default case for distributions we have are not building anymore since rospkg will be deployed across all ROS distros.

I agree with that assessment, but the alternative is to leave this know issue unresolved. The workaround for `/usr/share` being on the path is to manually set the env variable (for older distros that won't be updated) and to update the necessary ROS package for current distros. By comparison, the workaround for the other bug is to uninstall all of the packages.ros.org packages (including the rospkg), removing packages.ros.org from apt-get, then reinstalling rospkg. So, from my perspective this proposed action is a little messy, but also addresses the issue that is more difficult for users to deal with.

> I agree that it's not a great user experience, but there must be a way to keep /usr off the RPP.

In the current proposal there is. If you set `ROS_RPP_NO_DEFAULTS` in the environment then `/usr/share` will not be on the package path.

> It's very clear what your setting. There's no way for us to provide introspection to the default value because it's debian's requirement that it magically work without setting the value.

I agree with you, but we cannot get unsurprising behavior without setting an environment variable, but that is not allowed under 9.9 as @jspricke pointed out. So I think that between `/usr/share` always being implicitly on the RPP versus it being on the RPP until you set it, I'd prefer the former.

> I would advocate for using the RPP_NO_DEFAULTS and that would turn off the "default" which is only used when the RPP is set.

If that's the case, then you never need to set the RPP_NO_DEFAULTS, right? Because the `/opt/ros/kinetic` setup files will always set the RPP.

> If we require setting RPP_NO_DEFAULTS even if you've explicitly set the RPP in your setup script we will break every single existing environment setup file and example setup file because suddenly their environment will now have unknown packages from /usr unless they set RPP_NO_DEFAULTS.

Which setup files will have this problem? As long as they have sourced the `/opt/ros/kinetic/setup.*sh` file and they have `ros-core` installed, then `ROS_RPP_NO_DEFAULTS` will be set. There may be some cases, I just cannot think of them now.

I understand where you're coming from, but I still have a slight preference for `/usr/share` always being on the RPP, despite the fact that my first suggestion was to only add it if the RPP was unset.